### DISCUSSION TITLE
Post-migration script

### DISCUSSION THREAD

**Discussion by simon-sugarcrm** [amacneil/dbmate#531] [Support] [answered]
Hi, I'd like to run a privileges script after every migration. Is this possible with DBMate?

**Comment by dossy** [comment]
Executing a script after each migration is applied is not currently possible within dbmate, or at least I cannot imagine a way to do so.

What is the use case you're looking to implement?  Maybe there is an alternative solution that is possible.

**Reply by simon-sugarcrm** [reply]
Hi @dossy , I have a multi-tenant db (client per schema). DBMate runs migrations and builds new tenants very nicely. Currently I have to manually run a script to grant privileges to IT staff groups (Support, Dev, QA etc) to any new tenants. It would be great to have that automated. I know I could have a job that looks periodically for new schemas, but would like it all happen in one go if possible.

**Reply by dossy** [reply, marked as answer]
If you need a single script executed after executing dbmate, you could wrap your execution of dbmate into a shell script that executes dbmate then executes your additional script.  That would be the recommended approach.

If you need a script executed in between individual migration files as dbmate applies them in order, that's not something that dbmate is currently capable of.  If `dbmate up` took a `--limit` argument to control the number of migrations applied per run, it would be hypothetically possible to write a shell script that invoked `dbmate` in a loop, applying `--limit 1` migration file at a time, until there are no more migrations left to apply.

It sounds like you only need a single script executed once after dbmate applies 0 or more migrations, which would best be handled by a wrapper script that invokes dbmate then the script afterwards, which you could do today without any changes to dbmate.

**Reply by simon-sugarcrm** [reply]
Thanks ever so much dossy for taking the time to reply. That's really helpful and greatly appreciated.
